packages/core/src/interpreter.ts

```diff
@@ -417,6 +426,29 @@ export class Interpreter<
     this.doneListeners.add(listener);
     return this;
   }
+  /**
+   * Adds an error listener that is notified when an unhandled error occurs.
+   * @param errorListener The error listener
+   */
+  public onError(
+    errorListener: ErrorListener
+  ): Interpreter<TContext, TStateSchema, TEvent, TTypestate> {
+    this.errorListeners.add(errorListener);
+    return this;
+  }
+
+  private handleError(errorData: unknown): void {
+    const errorInstance =
```

Comment:
What should be the resolution here? Have error listeners treat received errors as `unknown`?

Reply:
Added `isError`... thinking more about what we can do here, we can make a `class SCXMLError extends Error { ... }` class that has:

```js
{
  name: 'error.execution', // or 'error.platform' etc.
  message: '...',
  // non-standard properties
  data: errorData, // comes from original error data,
  origin: '...'
}
```

And throw that instead. Then, the original error data can always be determined from `error.data` and we won't force developers to determine the type of an `unknown` error instead.

Reply:
For reasons that I don't remember subclassing native classes ain't working properly in ES5 (and we transpile down to it). Take this simple input:
```js
class MyError extends Error {}
```
and the TS-transpiled output (same happens with Babel etc) for it:
```js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var MyError = /** @class */ (function (_super) {
    __extends(MyError, _super);
    function MyError() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return MyError;
}(Error));
```
and now do the simple test:
```js
var err = new MyError()
console.log(err instanceof MyError) // oops! it's false 😬
```

---

> What should be the resolution here? Have error listeners treat received errors as unknown?

Either `unknown` or `any`. I think that error listeners should receive unwrapped errors, and not even "error events". If we want to provide extra metadata to error listeners then the second argument could contain it.

Error events should still be created so actual transitions could handle them. And I believe that `onError` shouldn't be called for handled error events. But this depends on the answers to the questions I've posted:
> It would be great to define (in the docs) what's the exact purpose of onError. What it should do and what it can't do. Is it useful only for reporting? Can it influence anyhow the machine's lifetime? This seems vastly important for users wanting to use this.

Reply:
Okay, no longer wrapping in error 👍